### PR TITLE
refactor: state constructor simplifications

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: src/Carousel.js [hunk 1]
```diff
@@ -120,17 +120,14 @@ const defaultProps = {
...
-    this.state = {
-      prevClasses: '',
-      currentClasses: 'active',
-    };
-    this.isUnmounted = false;
+  isUnmounted = false;
 
-    this.carousel = React.createRef();
-  }
+  carousel = React.createRef();
```
Comment (mxschmitt) [inline src/Carousel.js]: not sure if something like this is OK?

File: src/Modal.js [hunk 2]
```diff
@@ -183,14 +183,11 @@ function BackdropTransition(props) {
...
-    super(props, context);
+  state = { style: {} };
 
-    this.state = { style: {} };
-    this.modalContext = {
-      onHide: () => this.props.onHide(),
-    };
-  }
+  modalContext = {
+    onHide: () => this.props.onHide(),
```
Comment (mxschmitt) [inline src/Modal.js]: is the context here okay? (onHide method)

File: src/Navbar.js [hunk 3]
```diff
@@ -123,8 +123,8 @@ const defaultProps = {
 };
 
 class Navbar extends React.Component {
-  constructor(...args) {
-    super(...args);
+  constructor(props) {
```
Comment (mxschmitt) [inline src/Navbar.js]: changed, so that we have a consistency in the project.

File: src/Toast.js [hunk 4]
```diff
@@ -79,7 +79,7 @@ const Toast = ({
       };
     }
     return () => null;
-  }, [autohide, show]);
+  }, [autohide, delay, onClose, show]);
```
Comment (mxschmitt) [inline src/Toast.js]: Was a ESLint warning

File: src/Carousel.js (hunk 1 above)
Comment (bpas247) [reply src/Carousel.js]: maybe the assignments could go under cDM? probably doesn't matter too much with `this.isUnmounted`, since it's just a primitive.

Comment (taion) [reply src/Carousel.js]: this is fine

File: src/Modal.js (hunk 2 above)
Comment (taion) [reply src/Modal.js]: yup this works

File: src/Navbar.js (hunk 3 above)
Comment (taion) [reply src/Navbar.js]: why not make `state` an instance property here as well?

File: src/Toast.js (hunk 4 above)
Comment (taion) [reply src/Toast.js]: hmm, this was deliberate though. we didn't want to restart the timer in the event that `delay` changed. though `onClose` should probably take the latest value via a ref

File: src/Navbar.js (hunk 3 above)
Comment (mxschmitt) [reply src/Navbar.js]: because then the context of it is incorrect. So I have to create a wrap function:
`onToggle: () => this.handleToggle(),`

**Pull request merged**